test/functional/legacy/delete_spec.lua

```diff
@@ -98,4 +108,46 @@ describe('Test for delete()', function()
     eq(0, eval("delete('Xdir4/Xfile')"))
     eq(0, eval("delete('Xdir4', 'd')"))
   end)
+
+  it('complicated name delete', function()
+    source([[
+      call mkdir('Xcomplicated/[complicated-1 ]', 'p')
+      call mkdir('Xcomplicated/{complicated,2 }', 'p')
+      call writefile(['a', 'b'], 'Xcomplicated/Xfile')
+      call writefile(['a', 'b'], 'Xcomplicated/[complicated-1 ]/Xfile')
+      call writefile(['a', 'b'], 'Xcomplicated/{complicated,2 }/Xfile')
+    ]])
+
+    eq(1, eval("isdirectory('Xcomplicated')"))
+    eq(eval("['a', 'b']"), eval("readfile('Xcomplicated/Xfile')"))
+    eq(1, eval("isdirectory('Xcomplicated/[complicated-1 ]')"))
+    eq(eval("['a', 'b']"), eval("readfile('Xcomplicated/[complicated-1 ]/Xfile')"))
+    eq(1, eval("isdirectory('Xcomplicated/{complicated,2 }')"))
+    eq(eval("['a', 'b']"), eval("readfile('Xcomplicated/{complicated,2 }/Xfile')"))
+
+    eq(0, eval("delete('Xcomplicated', 'rf')"))
+    eq(0, eval("isdirectory('Xcomplicated')"))
```

Comment:
Confirmed that this works with the win32 build, plus the existing tests in here were passing, so that means the basic functionality of `isdirectory()` and `delete('dir')` were working on Windows. That suggests that there's still a remaining problem with `delete_recursive()` on Windows.